stable/spark/values.yaml

```diff
@@ -8,51 +8,60 @@ Spark:
 
 Master:
   Name: master
-  Image: "k8s.gcr.io/spark"
-  ImageTag: "1.5.1_v3"
+  Image: "dbanda/spark2.4"
+  ImageTag: "v3"
   Replicas: 1
   Component: "spark-master"
   Cpu: "100m"
-  Memory: "512Mi"
+  Memory: "1024Mi"
   ServicePort: 7077
   ContainerPort: 7077
   # Set Master JVM memory. Default 1g
   # DaemonMemory: 1g
   ServiceType: LoadBalancer
 
+WorkerLB:
+  Name: workerlb
+  Component: "workerlb"
+  ServicePort: 8080
+  ContainerPort: 8080
+
 WebUi:
   Name: webui
   ServicePort: 8080
   ContainerPort: 8080
 
 Worker:
   Name: worker
-  Image: "k8s.gcr.io/spark"
-  ImageTag: "1.5.1_v3"
-  Replicas: 3
+  Image: "dbanda/spark2.4"
+  ImageTag: "v3"
+  Replicas: 2
   Component: "spark-worker"
   Cpu: "100m"
-  Memory: "512Mi"
+  Memory: "2048Mi"
```

Comment:
This doesn't match the README, which probably would lead to some surprises from the end user.  Sorry, there doesn't seem to be a particularly friendly way to avoid this kind of nits because I've probably done it myself. :(

Also, my semi-opinionated perception of how charts are usually packaged is that you want to set them up so that an end-user could run the smallest practical version of things and then provide a values-production.yaml file with suggested changes for something that looks more like production, so you might want to consider the default values within that context.  Is it better to do 512Mi of memory and autoscaling off, but with something more practically sized easily accessible, understanding that this will be friendlier for your neighborhood Polite But Grumpy SRE Who Just Got Asked To Fire Up Spark By Their Boss And Is Going To Try Using Helm.

Reply:
Thanks for the feedback. In testing I found 512Mi too little for my typical spark jobs. OOM can sometimes be hard to catch in spark so I think it's better to liberally assign memory first then optimize from there. As for including values-production.yaml , I don't think my understanding of sparks configuration parameters is sufficient for me to adequetly suggest the best production values. The best values would vary widely based on the user's use case. It hard to suggest any production values with any confidence.